servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -733,16 +735,21 @@ private MoveBlockResult moveBlockInternal(long userId, long blockId,
         srcLocation = srcBlockMeta.getBlockLocation();
         srcFilePath = srcBlockMeta.getPath();
         blockSize = srcBlockMeta.getBlockSize();
+
+        // when the srcLocation equals the dstLocation, simply do nothing and return success
+        if (srcLocation.equals(dstLocation)) {
+          return new MoveBlockResult(true, blockSize, srcLocation, dstLocation);
+        }
       } finally {
         mMetadataReadLock.unlock();
       }
 
-      TempBlockMeta dstTempBlock =
-          createBlockMetaInternal(userId, blockId, newLocation, blockSize, false);
+      TempBlockMeta dstTempBlock = createBlockMetaInternal(userId, blockId, dstLocation,
+          blockSize, false);
       if (dstTempBlock == null) {
         return new MoveBlockResult(false, blockSize, null, null);
       }
-      dstLocation = dstTempBlock.getBlockLocation();
```

Comment:
In the original code, the above statement is redundant as this `dstLocation` and the original parameter `newLocation` should always be **the same**. So remove it and directly rename the third parameter from `newLocation` to `dstLocation` for clearness.